test/box-luatest/gh_8051_set_box_cfg_thru_env_test.lua

```diff
@@ -0,0 +1,23 @@
+local server = require('luatest.server')
+
+local t = require('luatest')
+local g = t.group()
+
+local function server_test_nested_table()
+    t.assert_equals(box.cfg.metrics.labels.alias, 'gh_8051')
+
+    t.assert_equals(box.cfg.log_modules, {['aaa'] = 'info', ['bbb'] = 'error'})
+end
+
+g.test_nested_table = function()
+    local env = {["TT_METRICS"] = '{"labels":{"alias":"gh_8051"},' ..
+                                  '"include":"all","exclude":["vinyl"]}',
+                 ["TT_LOG_MODULES"] = 'aaa=info,bbb=error'}
+
+    g.server = server:new{alias='default', env=env}
+    g.server:start()
+
+    g.server:exec(server_test_nested_table)
+
+    server:stop()
```

Comment:
If the test above fails, the server will not be stopped, which is not graceful. We usually use 'before test' and 'after test' helpers, they are readable and safe.

Reply:
Good point, thx. Fixed.